eclair-core/src/main/scala/fr/acinq/eclair/message/OnionMessages.scala

```diff
@@ -166,10 +167,11 @@ object OnionMessages {
   private def decryptOnion(privateKey: PrivateKey, packet: OnionRoutingPacket): Either[DropReason, DecodedOnionPacket] = {
     Sphinx.peel(privateKey, None, packet) match {
       case Right(p: Sphinx.DecryptedPacket) =>
-        MessageOnionCodecs.perHopPayloadCodec.decode(p.payload.bits) match {
-          case Attempt.Successful(DecodeResult(perHopPayload, _)) if p.isLastPacket => Right(DecodedOnionPacket(perHopPayload, None))
-          case Attempt.Successful(DecodeResult(perHopPayload, _)) => Right(DecodedOnionPacket(perHopPayload, Some(p.nextPacket)))
-          case Attempt.Failure(f) => Left(CannotDecodeOnion(f.message))
+        Try(MessageOnionCodecs.perHopPayloadCodec.decode(p.payload.bits)) match {
+          case Success(Attempt.Successful(DecodeResult(perHopPayload, _))) if p.isLastPacket => Right(DecodedOnionPacket(perHopPayload, None))
+          case Success(Attempt.Successful(DecodeResult(perHopPayload, _))) => Right(DecodedOnionPacket(perHopPayload, Some(p.nextPacket)))
+          case Success(Attempt.Failure(f)) => Left(CannotDecodeOnion(f.message))
+          case Failure(f) => Left(CannotDecodeOnion(s"unexpected exception in onion message codecs: ${f.getMessage}"))
```

Comment:
Looking at scodec's code, they do explicitly call `try`/`catch` in risky codecs (e.g. [StringCodec](https://github.com/scodec/scodec/blob/5d6d0fb24e9ea1eeec4aa7f604efe179ce5ce463/shared/src/main/scala/scodec/codecs/StringCodec.scala)). Shouldn't we do the same ourselves and ensure that those return proper `Attempt.Failure` instead of wrapping everything inside a `try`/`catch`? I think that `Attempt.fromTry` is designed for that purpose.